src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineControl
+    {
+        private IWpfTextView? GetLastActiveIWpfTextView()
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+
+            if (ErrorHandler.Failed(CodeWindow.GetLastActiveView(out var textView)))
+                return null;
+
+            return EditorAdaptersFactoryService.GetWpfTextView(textView);
+        }
+
+        /// <summary>
+        /// Starts a new task to compute the UI model.
+        /// </summary>
+        private void StartComputeUIModelTask()
+        {
+            // 'true' value is unused.  this just signals to the queue that we have work to do.
```

Comment:
i see. please doc this on the fields to make it clear we don't care about that `bool` parameter.  tahnks~

Reply:
Is it worth just adding another helper like https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue%600.cs that doesn't require all these comments?

Reply:
Resolving this comment since I added docs. I don't think I'll be adding this helper now (as the code copies the pattern found in `NavigationBarController_ModelComputation.cs`)